### PR TITLE
[WIP] Testing PR for sync upstream v1.9 changes to our branch v1.7

### PR DESCRIPTION
**Pull request by atheo89**
## Description
This PR is a testing PR that tries to sync kubeflow/v1.7-branch with the upstream kubeflow/1.9-branch.

Used only to check the GHA tests

created with following commands:

```
git remote add kfbf-upstream  https://github.com/kubeflow/kubeflow.git
git fetch --all && git pull --all
git remote -v
  kfbf-upstream   https://github.com/kubeflow/kubeflow.git (fetch)
  kfbf-upstream   https://github.com/kubeflow/kubeflow.git (push)
  origin  https://github.com/atheo89/kubeflow.git (fetch)
  origin  https://github.com/atheo89/kubeflow.git (push)
  upstream        https://github.com/opendatahub-io/kubeflow.git (fetch)
  upstream        https://github.com/opendatahub-io/kubeflow.git (push)
git checkout -b sync-test2 upstream/1.7-branch
git pull  kfbf-upstream v1.9-branch --no-rebase
Resolve conflicts
git add . && git commit
git push origin synh-test2
```

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by atheo89** [comment]
One thing I noticed, which is likely the reason for the code-static analysis tests failing, is that the go.mod and go.sum files were using older versions compared to our current changes. I accepted the incoming changes, but perhaps I should have used ours instead.

**Comment by jstourac** [comment]
Do we plan to fetch the code from upstream 1.9 into our 1.7 branch or is this just and only for the testing purpose?
How much work it would be if we would took the 1.9 upstream branch and apply our custom changes on top of it instead? (ideally pushing these changes into the upstream sometime later).

**Comment by atheo89** [comment]
> Do we plan to fetch the code from upstream 1.9 into our 1.7 branch or is this just and only for the testing purpose?

Hi @jstourac, after discussing with @harshad16, we've decided to proceed with the first option you mentioned: fetching the code from upstream 1.9 into our 1.7 branch

>How much work it would be if we would took the 1.9 upstream branch and apply our custom changes on top of it instead? (ideally pushing these changes into the upstream sometime later).

It would definitely require a significant amount of work, and there’s a risk we might overlook some critical work or changes during the process.

Additionally, we aim to keep our master branch synced with the real upstream master to maintain alignment from now on.

**Comment by harshad16** [comment]
> One thing I noticed, which is likely the reason for the code-static analysis tests failing, is that the go.mod and go.sum files were using older versions compared to our current changes. I accepted the incoming changes, but perhaps I should have used ours instead.

yes , we should use all the upgrades that we have done, and only accept the other changes they are bringing.
that what we need to test about.

**Comment by harshad16** [comment]
> Do we plan to fetch the code from upstream 1.9 into our 1.7 branch or is this just and only for the testing purpose? How much work it would be if we would took the 1.9 upstream branch and apply our custom changes on top of it instead? (ideally pushing these changes into the upstream sometime later).


My understanding, the approach 2 would require cherry-picks of the content or rebase, which is technically a overwrite.

odh-1.7 branch: changes from us + upstream
kfp-1.9 branch: changes from upstream.

Approach 1: 
- cut-off a branch v1.9 from odh-v1.7 
- Pull changes of upstream v1.9 to this branch.

This way we only pull new changes that are done on upstream without any cherry-picks.

Approach 2:
- Pull upstream branch v1.9 to odh-v1.9.
- Cherry picks changes from v1.7 branch that are done over.

significant cherry-pick would need to be done on this.
please do share if you believe i missed something , we could perhaps go with another approach.

The current approach allows us to keep the new changes on top, we can move forward.

File: components/notebook-controller/controllers/notebook_controller.go [hunk 1]
```diff
@@ -532,8 +544,14 @@ func generateVirtualService(instance *v1beta1.Notebook) (*unstructured.Unstructu
...
-	if err := unstructured.SetNestedStringSlice(vsvc.Object, []string{"*"}, "spec", "hosts"); err != nil {
-		return nil, fmt.Errorf("set .spec.hosts error: %v", err)
+
+	istioHost := os.Getenv("ISTIO_HOST")
+	if len(istioHost) == 0 {
+		istioHost = "*"
+	}
+	if err := unstructured.SetNestedStringSlice(vsvc.Object, []string{istioHost}, "spec", "hosts"); err != nil {
+		return nil, fmt.Errorf("Set .spec.hosts error: %v", err)
+
```
Comment (jiridanek) [inline components/notebook-controller/controllers/notebook_controller.go:554]: I'm doubtful about this. Do we want to pull in anything Istio-related? The changes in manifests do set `ISTIO_HOST`.

From a quick look, it should not be harmful, so we may just as well have this.

**Comment by atheo89** [comment]
Thank you guys for your review so far, I have opened a new PR here -> https://github.com/opendatahub-io/kubeflow/pull/385 

Closing this

File: components/notebook-controller/controllers/notebook_controller.go (hunk 1 above)
Comment (atheo89) [reply components/notebook-controller/controllers/notebook_controller.go:554]: Hey Jiri! Yes, they have added two new configmap envs the `ISTIO_HOST` and `CLUSTER_DOMAIN` on components/notebook-controller/config/manager/manager.yaml file
I don't think that they are harmful as well. @harshad16 wdyt?

**Pull request closed without merging**